examples/user_guide/8_Geography.ipynb

```diff
@@ -272,26 +279,58 @@
    "outputs": [],
    "source": [
     "from functools import partial\n",
-    "\n",
-    "def heights(locations, min_val, max_val, height):\n",
-    "    out = np.zeros(len(locations))\n",
-    "    for i, (x, y) in enumerate(locations):\n",
-    "        val = terrain.data[y, x]\n",
-    "        out[i] = height if (val > min_val and val < max_val) else 0\n",
+    "from datashader.utils import ngjit\n",
+    "\n",
+    "@ngjit\n",
+    "def height_func(locations, elevation_source, min_elevation, max_elevation, height=20):\n",
+    "    '''Example user-defined height_func for bump mapping which varies bump height by elevation thresholding\n",
+    "    \n",
+    "    Returns\n",
+    "    -------\n",
+    "    out: 1D array of heights for each location\n",
+    "    \n",
+    "    '''\n",
+    "    num_bumps = locations.shape[0]\n",
+    "    out = np.zeros(num_bumps, dtype=np.uint16)\n",
+    "    for r in range(0, num_bumps):\n",
+    "        loc = locations[r]\n",
+    "        x = loc[0]\n",
+    "        y = loc[1]\n",
+    "        val = elevation_source[y, x]\n",
+    "        if val >= min_elevation and val < max_elevation:\n",
+    "            out[r] = height\n",
     "    return out\n",
     "\n",
-    "T = 100000\n",
+    "low_elevation_tree_heights = partial(height_func, elevation_source=terrain.data, min_elevation=1000, max_elevation=2000, height=10)\n",
+    "mid_elevation_tree_heights = partial(height_func, elevation_source=terrain.data, min_elevation=2000, max_elevation=3000, height=5)\n",
+    "high_elevation_tree_heights = partial(height_func, elevation_source=terrain.data, min_elevation=3000, max_elevation=5000, height=5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's use our custom `height_func` with `dsgeo.bump`..."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TREE_COUNT = int(3e5)\n",
     "\n",
-    "trees = dsgeo.bump(W, H, count=T, height_func=\n",
-    "                   partial(heights, min_val=1300,  max_val=1700, height=20))\n",
+    "trees = dsgeo.bump(W, H, count=TREE_COUNT, height_func=low_elevation_tree_heights)\n",
+    "trees += dsgeo.bump(W, H, count=TREE_COUNT // 2, height_func=mid_elevation_tree_heights)\n",
+    "trees += dsgeo.bump(W, H, count=TREE_COUNT // 3, height_func=high_elevation_tree_heights)\n",
```

Comment:
(It's great to have more complex examples for people diving into it, but first there needs to be some simple, easy to read and adapt example for everyone else. I think a long example like this one just makes readers' eyes glaze over and they skip the whole page of code, instead of actually reading it.)